src/remoteConnector.ts

```diff
@@ -900,12 +903,76 @@ export default class RemoteConnector extends Disposable {
 		}
 	}
 
-	private startHeartBeat(accessToken: string, connectionInfo: SSHConnectionParams) {
+	private async startHeartBeat(accessToken: string, connectionInfo: SSHConnectionParams, gitpodVersion: GitpodVersion) {
 		if (this.heartbeatManager) {
 			return;
 		}
 
 		this.heartbeatManager = new HeartbeatManager(connectionInfo.gitpodHost, connectionInfo.workspaceId, connectionInfo.instanceId, accessToken, this.logger, this.telemetry);
+
+		// gitpod remote extension installation is async so sometimes gitpod-desktop will activate before gitpod-remote
+		// let's try a few times for it to finish install
+		try {
+			await retry(async () => {
+				await vscode.commands.executeCommand('__gitpod.cancelGitpodRemoteHeartbeat');
+			}, 3000, 15);
+			this.telemetry.sendTelemetryEvent('vscode_desktop_heartbeat_state', { enabled: String(true), gitpodHost: connectionInfo.gitpodHost, workspaceId: connectionInfo.workspaceId, instanceId: connectionInfo.instanceId, gitpodVersion: gitpodVersion.raw });
+		} catch {
+			this.logger.error(`Could not execute '__gitpod.cancelGitpodRemoteHeartbeat' command`);
+			this.telemetry.sendTelemetryEvent('vscode_desktop_heartbeat_state', { enabled: String(false), gitpodHost: connectionInfo.gitpodHost, workspaceId: connectionInfo.workspaceId, instanceId: connectionInfo.instanceId, gitpodVersion: gitpodVersion.raw });
+		}
+	}
+
+	private async initializeRemoteExtensions() {
+		let syncData: { ref: string; content: string } | undefined;
+		try {
+			syncData = await this.settingsSync.readResource(SyncResource.Extensions);
+		} catch (e) {
+			if (e instanceof NoSyncStoreError) {
+				const action = 'Settings Sync: Enable Sign In with Gitpod';
+				const result = await vscode.window.showInformationMessage(`Couldn't initialize remote extensions, Settings Sync with Gitpod is required.`, action);
+				if (result === action) {
+					vscode.commands.executeCommand('gitpod.syncProvider.add');
+				}
+			} else if (e instanceof NoSettingsSyncSession) {
+				const action = 'Enable Settings Sync';
+				const result = await vscode.window.showInformationMessage(`Couldn't initialize remote extensions, please enable Settings Sync.`, action);
+				if (result === action) {
+					vscode.commands.executeCommand('workbench.userDataSync.actions.turnOn');
+				}
+			} else {
+				this.logger.error('Error while fetching settings sync extension data', e);
+			}
+			return;
+		}
+
+		const syncDataContent = parseSyncData(syncData.content);
+		if (!syncDataContent) {
+			this.logger.error('Error while parsing sync data');
+			return;
+		}
+
+		let extensions: ISyncExtension[];
+		try {
+			extensions = JSON.parse(syncDataContent.content);
+		} catch {
```

Comment:
i think it is important always log underlying issues, please don't skip error

Reply:
There's already a log for this